documentation/FIR-MLIR-dialect.md

```diff
@@ -0,0 +1,204 @@
+# Investigating FIR as an MLIR Dialect
+
+## Introduction
+
+With the recent availability of the extensible MLIR framework as open-source, we've been investigating the potential to use MLIR as a substrate upon which to build a FIR dialect. This document will explain the motivations and the shape of this subproject using MLIR to define and implement a high-level operational representation of Fortran. (See also _Design: Fortran IR_ at [https://github.com/flang-compiler/f18/blob/master/documentation/FortranIR.md](https://github.com/flang-compiler/f18/blob/master/documentation/FortranIR.md) for more information.)
```

Comment:
The name of the file (FIR-MLIR-dialect.md) implies that this is the documentation for the FIR dialect instead of an investigation.  I'm ok with that name if we intend to update this document as the FIR dialect matures and at that time remove phrases such as "recent'.

Reply:
I think of this document as more of the preliminary design than a one-off investigation.